teamcity/resource_build_trigger_schedule.go

```diff
@@ -210,6 +211,13 @@ func resourceBuildTriggerScheduleCreate(d *schema.ResourceData, meta interface{}
 		}
 	}
 
+	if strings.EqualFold(schedule, "weekly") {
+		triggerWeekday, ok := d.GetOk("weekday")
+		if !ok || triggerWeekday == nil || triggerWeekday.(string) == "" {
```

Comment:
I don't think it'll ever be possible for triggerWeekday to equal `nil` if `!ok` is the case.

Reply:
For QoL, what do you think of testing for the 7 valid days here -- e.g. with a 2-case `switch` statement on the lowercase value `triggerWeekday`; one `case "sunday", ...:` and another `default:` for the failure case)

That way, typos and such will also be caught during `terraform validate/plan` (i.e. the "Test CI Config" check), rather than being accepted & only failing when the apply is attempted/rejected by TeamCity?

Reply:
> I don't think it'll ever be possible for triggerWeekday to equal nil if !ok is the case.

Actually, it doesn't look like you need to check for anything other than `!ok`, if I'm reading the [implementation of GetOk](https://github.com/hashicorp/terraform/blob/v0.12.0/helper/schema/resource_data.go#L93) right.

Can this code be changed to only check for `!ok`?